util/grpcutil/stats_filter.go

```diff
@@ -0,0 +1,47 @@
+package grpcutil
+
+import (
+	"context"
+
+	"google.golang.org/grpc/stats"
+)
+
+type contextKey int
+
+const filterContextKey contextKey = iota
+
+type StatsFilterFunc func(info *stats.RPCTagInfo) bool
+
+func StatsFilter(h stats.Handler, fn StatsFilterFunc) stats.Handler {
+	return &statsFilter{
+		inner:  h,
+		filter: fn,
+	}
+}
+
+type statsFilter struct {
+	inner  stats.Handler
+	filter func(info *stats.RPCTagInfo) bool
+}
+
+func (s *statsFilter) TagRPC(ctx context.Context, info *stats.RPCTagInfo) context.Context {
+	if s.filter(info) {
+		return context.WithValue(ctx, filterContextKey, struct{}{})
+	}
+	return s.inner.TagRPC(ctx, info)
+}
+
+func (s *statsFilter) HandleRPC(ctx context.Context, rpcStats stats.RPCStats) {
+	if ctx.Value(filterContextKey) != nil {
```

Comment:
I think I misunderstood the extent of this deprecation before. It seemed to me that the whole interceptors concept in grpc was deprecated and replaced with "stats" what left me confused as they don't seem to have similar capabilities.

But looks like that only thing that is deprecated is the interceptor implementations for otel library. In that case I think the only change I would like is to make it more clear that the "stats_filter" is the filter only for otel handlers. Maybe call it "otelfilter" or even put it under "util/tracing". As we don't actually use any of the "stats" feature and it just seems to be how the otel interceptor now exposes itself then better to avoid that naming as much as possible.

Reply:
Updated to move this into the `tracing` package as something that's unexported and now we just create the stats handlers from the tracing package.